File: Squares With Three/Squares With Three.dart
// Time Complexity: O(n)
// Space Complexity: O(1)
int squaresWithThree(int n) {
  int count = 0;
  for (int i = 1; i <= n; i++) {
    if ((i * i).toString().contains('3')) {
      count++;
    }
  }
  return count;
}

void main() {
  int n = 10;
  print('Count: [32m${squaresWithThree(n)}');
}

File: Squares With Three/Squares With Three.rs
// Time Complexity: O(n)
// Space Complexity: O(1)

fn squares_with_three(n: u32) -> u32 {
    let mut count = 0;
    for i in 1..=n {
        if (i * i).to_string().contains('3') {
            count += 1;
        }
    }
    count
}

fn main() {
    let n = 10;
    println!("Count: {}", squares_with_three(n));
}
